Cython/Compiler/ExprNodes.py

```diff
@@ -7014,27 +7014,35 @@ def analyse_as_type_attribute(self, env):
                     else:
                         # Create a temporary entry describing the C method
                         # as an ordinary function.
-                        if entry.func_cname and not hasattr(entry.type, 'op_arg_struct'):
-                            cname = entry.func_cname
-                            if entry.type.is_static_method or (
-                                    env.parent_scope and env.parent_scope.is_cpp_class_scope):
-                                ctype = entry.type
-                            elif type.is_cpp_class:
-                                error(self.pos, "%s not a static member of %s" % (entry.name, type))
-                                ctype = PyrexTypes.error_type
+                        def create_ubcm_entry(entry):
```

Comment:
I quite like nested functions/classes but most of the time when I try to use then I get encouraged not to for alleged performance reasons :'(

I wonder if this would be better as a non-nested function with `env` (I think that's the only captured variable?) passed as an argument?

Reply:
To quote the Zen of Python:

> Flat is better than nested.

So I guess I made the wrong choice here :) I'll pull this out into a separate method.

Reply:
Fixed.

Reply:
> I quite like nested functions/classes but most of the time when I try to use then I get encouraged not to for alleged performance reasons

Don't let them fool you. Readability counts. More than performance, most of the time.